engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/job/ExecuteExpressionJob.java

```diff
@@ -55,4 +55,9 @@ public Executable run(RuntimeContext ctx) {
           });
     }
   }
+
+  @Override
+  public boolean equalsTo(UniqueJob<?> that) {
+    return that instanceof ExecuteExpressionJob;
```

Comment:
Equality based on class of the job? E.g. only one job may be present regardless of the actual `oneShotExpression`?

It seems to me that the engine is trying to _handle cancellation_ without understanding what is being cancelled. I think only the IDE can say that a `oneShotExpression` and its `ExecuteExpressionJob` shall be cancelled - the engine cannot decide on that.